app/helpers/application_helper.rb

```diff
@@ -58,6 +58,50 @@ def sample_types_page_expermient?
       !@experiment.nil?
   end
 
+  def smart_annotation_notification(options = {})
+    title = options.fetch(:title) { :title_must_be_present }
```

Comment:
if we get the field like title = options[:title] and you call this smart_annotation_notification function without passing the title in the function call the title will a nil value and later when you cast it into a string it will be an empty string. In the case you use fetch the function will blow up and you will get the :title_must_be_present error message. For the old_text in this case I don't use fetch because the param is optional. I only convert ti to an empty string if the param is nil so I can call gsub on it.